rest/tests/test_rest.py

```diff
@@ -0,0 +1,579 @@
+            self.repo.slug, new_vocab_slug, {'name': 'rename'})
+        new_vocab_slug = output_dict['slug']
+        self.assertEqual('rename',
+                         Vocabulary.objects.get(slug=new_vocab_slug).name)
+
+        # put with missing slug
+        self.put_vocabulary(self.repo.slug, "missing slug", input_dict,
+                            expected_status=HTTP_404_NOT_FOUND)
+
+        output_dict = self.put_vocabulary(
+            self.repo.slug, new_vocab_slug, input_dict)
+        new_vocab_slug = output_dict['slug']
+        self.assertEqual(input_dict['name'],
+                         Vocabulary.objects.get(slug=new_vocab_slug).name)
+        # never created a duplicate
+        self.assertEqual(1, Vocabulary.objects.count())
+
+        # try deleting a vocabulary
+        self.delete_vocabulary(self.repo.slug, new_vocab_slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(1, vocabularies['count'])
+
+        # test missing repository
+        self.create_vocabulary("missing", input_dict,
+                               expected_status=HTTP_404_NOT_FOUND)
+        self.get_vocabularies("missing", expected_status=HTTP_404_NOT_FOUND)
+        self.get_vocabulary("missing", "missing",
+                            expected_status=HTTP_404_NOT_FOUND)
+
+        # as anonymous
+        self.logout()
+        self.get_vocabularies("missing", expected_status=HTTP_403_FORBIDDEN)
+        self.get_vocabulary("missing", "missing",
+                            expected_status=HTTP_403_FORBIDDEN)
+
+    def test_other_vocabulary(self):
+        """Test that two repositories have two different vocabularies"""
+
+        # missing fields
+        input_dict = {
+            'name': 'name',
+            'description': 'description',
+        }
+        self.create_vocabulary(self.repo.slug, input_dict,
+                               expected_status=HTTP_400_BAD_REQUEST)
+
+        input_dict = {
+            'name': 'name',
+            'description': 'description',
+        }
+        other_repo_dict = self.create_repository(input_dict)
+
+        vocab1_dict = {
+            'name': 'name',
+            'description': 'description',
+            'required': True,
+            'weight': 1000,
+            'vocabulary_type': 'f',
+        }
+
+        resp = self.client.post(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=self.repo.slug,
+                repo_base=REPO_BASE,
+            ),
+            vocab1_dict,
+        )
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+
+        vocab1_dict = self.create_vocabulary(self.repo.slug, vocab1_dict)
+
+        # test duplicate vocabulary
+        self.create_vocabulary(self.repo.slug, vocab1_dict,
+                               expected_status=HTTP_400_BAD_REQUEST)
+
+        vocab2_dict = {
```

Comment:
For clarity, it would be better to re-use `vocab1_dict` from above (maybe change the name to `vocab_dict`), and just set `vocab_dict["name"] = "name 2"` so it's obvious that what's being tested here is a unique name.